grails-app/domain/org/pih/warehouse/order/OrderItem.groovy

```diff
@@ -286,6 +290,20 @@ class OrderItem implements Serializable, Comparable<OrderItem> {
         return sortOrder
     }
 
+    def getSubmittedInvoiceItems() {
+        return InvoiceItem.list().findAll { it.orderItem == this && it.invoice.dateSubmitted }
+    }
+
+    Integer getQuantityInvoicedInStandardUom() {
+        return submittedInvoiceItems.sum { InvoiceItem invoiceItem ->
+            invoiceItem?.quantity
```

Comment:
Same as above. This would be the one place where I wouldn't mind leaving this as-is, but this can definitely be done in a single query.